pkg/util/mathutil/math.go

```diff
@@ -110,3 +110,31 @@ func Divide2Batches(total, batches int) []int {
 	}
 	return result
 }
+
+// Divide2Batches divies 'items' into 'batches', and returns the items in each batch.
+// The size of each batch is determined by Divide2BatchSizes.
+func Divide2Batches[T any](items []T, batches int, minBatchCnt, maxBatchCnt int) [][]T {
+	if len(items) == 0 {
+		return [][]T{}
+	}
+	if minBatchCnt > 0 {
+		maxBatches := len(items) / minBatchCnt
+		batches = min(batches, maxBatches)
+	}
+	batchSizes := Divide2BatchSizes(len(items), batches)
+	result := make([][]T, 0, batches)
+	start := 0
+	for start < len(items) {
+		for i := 0; i < len(batchSizes); i++ {
+			if batchSizes[i] == 0 {
+				// if current batch is 0, all the rest should be 0
+				break
+			}
+			count := min(batchSizes[i], maxBatchCnt)
```

Comment:
can we consider `maxBatchCnt` when calculating the number of batches we want to split? `Divide2BatchSizes` already done the work to split, we don't need to split it again here